test/index.spec.js

```diff
@@ -148,6 +148,19 @@ describe('main', () => {
         return reader._startKinesis('stream name', {})
       })
 
+      it('emits connected event when successfully connected', () => {
+        client.describeStream = AWSPromise.resolve({StreamDescription: {Shards: [{ShardId: 'shard id'}]}})
+        client.getShardIterator = AWSPromise.resolve({ShardIterator: 'shard iterator'})
+
+        sinon.stub(main.KinesisStreamReader.prototype, 'readShard')
+        const reader = new main.KinesisStreamReader(client, 'stream name', {foo: 'bar'})
+
+        return new Promise((resolve) => {
+          reader.once('connected', resolve)
+          reader._startKinesis()
```

Comment:
I can see why it's structured this way, and testing events in Mocha is really messy, but this test structure is hard to follow. I may revisit this in the future.

Reply:
For sure! There's some indirection here that I wasn't sure how to avoid. I don't feel attached to this code in any way. :)